extensions/core/interfaces/code/interface.vue

```diff
@@ -0,0 +1,161 @@
+<template>
+  <div class="interface-code">
+    <codemirror ref="cm"
+                :value="options.placeholder"
+                :options="cmOptions"
+                @ready="onCmReady"
+                @focus="onCmFocus"
+                @input="onCmCodeChange"
+    ></codemirror>
+    <div class="clipboard-btn">
+      <button class="btn" @click="$emit('input', code)">
+        <i class="material-icons">assignment_turned_in</i>
+      </button>
+    </div>
+  </div>
+</template>
+
+<script>
+
+// require component
+import { codemirror } from 'vue-codemirror';
+
+// require styles
+import 'codemirror/lib/codemirror.css'
+
+// language
+import 'codemirror/mode/javascript/javascript.js'
+
+// theme css
+// import 'codemirror/theme/monokai.css'
+
+// require active-line.js
+import'codemirror/addon/selection/active-line.js'
+
+// styleSelectedText - (styleSelectedText and css class .CodeMirror-selectedtext)
+import'codemirror/addon/selection/mark-selection.js'
+import'codemirror/addon/search/searchcursor.js'
+
+// hint - (extraKeys and hintOptions)
+import'codemirror/addon/hint/show-hint.js'
+import'codemirror/addon/hint/show-hint.css'
+import'codemirror/addon/hint/javascript-hint.js'
+import'codemirror/addon/selection/active-line.js'
+
+// highlightSelectionMatches - (highlightSelectionMatches)
+import'codemirror/addon/scroll/annotatescrollbar.js'
+import'codemirror/addon/search/matchesonscrollbar.js'
+import'codemirror/addon/search/searchcursor.js'
+import'codemirror/addon/search/match-highlighter.js'
+
+// keyMap - (matchBrackets)
+import'codemirror/mode/clike/clike.js'
+import'codemirror/addon/edit/matchbrackets.js'
+import'codemirror/addon/comment/comment.js'
+import'codemirror/addon/dialog/dialog.js'
+import'codemirror/addon/dialog/dialog.css'
+import'codemirror/addon/search/searchcursor.js'
+import'codemirror/addon/search/search.js'
+import'codemirror/keymap/sublime.js'
+
+// foldGutter - (foldGutter and gutters)
+import'codemirror/addon/fold/foldgutter.css'
+import'codemirror/addon/fold/brace-fold.js'
+import'codemirror/addon/fold/comment-fold.js'
+import'codemirror/addon/fold/foldcode.js'
+import'codemirror/addon/fold/foldgutter.js'
+import'codemirror/addon/fold/indent-fold.js'
+import'codemirror/addon/fold/markdown-fold.js'
+import'codemirror/addon/fold/xml-fold.js'
+
+import './code.css';
+import mixin from '../../../mixins/interface';
+
+export default {
+  name: "interface-code",
+  mixins: [mixin],
+  data () {
+    return {
+      code: "",   
```

Comment:
There's a `npm run lint` and a matching `npm run fix` command to make the code lint free according to the same specs as the rest of the app 🙂